### PR TITLE
Add private next-token suppression setting

### PR DESCRIPTION
**Pull request by vikhyat**
## Summary
- Add private `_suppress_next_token_ids` request plumbing through the engine and scheduler.
- Apply the setting only to the first generated token, after existing skill constraints.
- Preserve baseline sampled-token logprobs by overwriting affected first-token logprobs from pre-private-suppression logits.

## Tests
- `PYTHONDONTWRITEBYTECODE=1 PYTHONPATH=/home/vikhyat/code/scratch/worktrees/kestrel-suppress-next-token:/home/vikhyat/code/kestrel-proprietary/kestrel-kernels/python /home/vikhyat/code/kestrel/.venv/bin/pytest -p no:cacheprovider tests/test_engine.py tests/scheduler/test_logprobs.py`

### CODE REVIEW

**Comment by vikhyat** [comment]
@codex review

File: kestrel/scheduler/scheduler.py [hunk 1]
```diff
@@ -1301,6 +1303,61 @@ def _sample_batch(
...
+                seq.finalized
+                or not suppress
+                or seq.skill_state.token_count != 0
+            ):
+                continue
+            for token_id in suppress:
+                if token_id < 0 or token_id >= vocab_size:
+                    raise ValueError(
+                        "_suppress_next_token_ids contains token id "
+                        f"{token_id}, but vocab size is {vocab_size}"
```
[the remark this answers is not in the document]
Comment (vikhyat) [reply kestrel/scheduler/scheduler.py]: Fixed by moving `_suppress_next_token_ids` validation to request build/admission, where exceptions fail only the offending request. `_sample_batch` now just applies already-validated first-token suppression and no longer raises these request-scope validation errors.

**Comment by vikhyat** [comment]
@codex review

File: kestrel/scheduler/scheduler.py [hunk 2]
```diff
@@ -1333,8 +1363,33 @@ def _sample_batch(
...
+    def _selected_logprobs_from_logits(
+        logits: Tensor,
+        sampled_ids: Tensor,
+        temperatures: Tensor,
+    ) -> Tensor:
+        temps = torch.clamp(temperatures, min=0.0)
+        effective_temp = torch.where(
+            temps > _SAMPLING_EPS,
+            temps,
+            torch.ones_like(temps),
```
[the remark this answers is not in the document]
Comment (vikhyat) [reply kestrel/scheduler/scheduler.py:1387]: I think this is a false positive as written. `sample_step_from_logits` does not sample from a sharp low-temperature distribution for `temp <= 1e-6`; that threshold is defined as greedy in the dispatcher contract, the torch fallback, and the CUDA softmax-greedy path. `_selected_logprobs_from_logits` preserves the actual row temperature for `temp > 1e-6`, and for `temp <= 1e-6` it follows the greedy baseline-logprob convention used by this overwrite path.
